backend/src/resolvers/posts.resolver.ts

```diff
@@ -0,0 +1,79 @@
+import PostDatasource from "../datasources/post.datasource";
+import { JwtPayload } from "../jwt-payload";
+import Post from "../entities/post.entity";
+import UserDatasource from "../datasources/user.datasource";
+
+export const query = {
+  posts: async (_source, _args, ctx) => {
+    const postDatasource: PostDatasource = ctx.dataSources.postDatasource;
+    const posts = await postDatasource.getPosts();
+
+    return posts.map((post) => {
+      return {
+        ...post,
+        votes: post.voters.length,
+      };
+    });
+  },
+};
+
+export const properties = {
+  author: async (parent: Post, _args, ctx) => {
+    const userDatasource: UserDatasource = ctx.dataSources.userDatasource;
+    return await userDatasource.getUser(parent.user);
+  },
+};
+
+export const mutation = {
+  write: async (_, args, ctx) => {
+    const jwtPayload: JwtPayload = ctx.user;
+    const postDatasource: PostDatasource = ctx.dataSources.postDatasource;
+    const post: Partial<Post> = {
+      title: args.post.title,
+      user: jwtPayload.id,
```

Comment:
:star: For assigning the   authenticated user to a post in write and upvote mutations.